Combined version is not compatible with preprocessor

The combined version introduced in #2031 to fix #1609 only works in a C++ context. Trying to use it in a preprocessor context will give errors.

To test:

Modify the unit test in `source/MaterialXTest/MaterialXCore/CoreUtil.cpp` to add:

```
TEST_CASE("Version comparison", "[coreutil]")
{
    // Test for version comparison (C++ context)
    REQUIRE(MATERIALX_VERSION_INDEX > MATERIALX_GENERATE_INDEX(1, 38, 8));

    // Test for version comparison (preprocessor context)
#if MATERIALX_VERSION_INDEX > MATERIALX_GENERATE_INDEX(1, 38, 8)
    REQUIRE(TRUE);
#else
    REQUIRE(FALSE);
#endif
}

```

Results in the following compilation errors:

Microsoft:
```
fatal error C1012: unmatched parenthesis: missing ')'
```

GCC:
```
error: missing binary operator before token "("
    8 | #define MATERIALX_GENERATE_INDEX(major, minor, build) ((((uint32_t)(major)) << 22U) | (((uint32_t)(minor)) << 12U) | ((uint32_t)(build)))
      |                                                                    ^
note: in expansion of macro 'MATERIALX_GENERATE_INDEX'
   11 | #define MATERIALX_VERSION_INDEX MATERIALX_GENERATE_INDEX(MATERIALX_MAJOR_VERSION, MATERIALX_MINOR_VERSION, MATERIALX_BUILD_VERSION)
      |                                 ^~~~~~~~~~~~~~~~~~~~~~~~
note: in expansion of macro 'MATERIALX_VERSION_INDEX'
   15 | #if MATERIALX_VERSION_INDEX > MATERIALX_GENERATE_INDEX(1,38,7)
      |     ^~~~~~~~~~~~~~~~~~~~~~~
```

Clang:
```
error: token is not a valid binary operator in a preprocessor subexpression
   15 | #if MATERIALX_VERSION_INDEX > MATERIALX_GENERATE_INDEX(1,38,7)
      |     ^~~~~~~~~~~~~~~~~~~~~~~
note: expanded from macro 'MATERIALX_VERSION_INDEX'
   11 | #define MATERIALX_VERSION_INDEX MATERIALX_GENERATE_INDEX(MATERIALX_MAJOR_VERSION, MATERIALX_MINOR_VERSION, MATERIALX_BUILD_VERSION)
      |                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
note: expanded from macro 'MATERIALX_GENERATE_INDEX'
    8 | #define MATERIALX_GENERATE_INDEX(major, minor, build) ((((uint32_t)(major)) << 22U) | (((uint32_t)(minor)) << 12U) | ((uint32_t)(build)))
      |                                                          ~~~~~~~~~~^
1 error generated.
```

Which is a complete head scratcher on Windows, but makes sense in GCC and Clang: the preprocessor does not know how to cast integer values to `uint32_t` and does not care about potential shift overflows.

Fix is to remove the cast to `uint_32`.

--- a/source/MaterialXCore/Library.h
+++ b/source/MaterialXCore/Library.h
@@ -21,8 +21,8 @@
 
 #include <MaterialXCore/Generated.h>
 
-#define MATERIALX_GENERATE_INDEX(major, minor, build) \
-    ((((uint32_t)(major)) << 22U) | (((uint32_t)(minor)) << 12U) | ((uint32_t)(build)))
+#define MATERIALX_GENERATE_INDEX(major, minor, build) (((major) << 22U) | ((minor) << 12U) | (build))
+
 #define MATERIALX_VERSION_INDEX \
     MATERIALX_GENERATE_INDEX(MATERIALX_MAJOR_VERSION, MATERIALX_MINOR_VERSION, MATERIALX_BUILD_VERSION)
 

--- a/source/MaterialXTest/MaterialXCore/CoreUtil.cpp
+++ b/source/MaterialXTest/MaterialXCore/CoreUtil.cpp
@@ -12,8 +12,15 @@ namespace mx = MaterialX;
 
 TEST_CASE("Version comparison", "[coreutil]")
 {
-    // Test for version comparison
+    // Test for version comparison (C++ context)
     REQUIRE(MATERIALX_VERSION_INDEX > MATERIALX_GENERATE_INDEX(1, 38, 8));
+
+    // Test for version comparison (preprocessor context)
+#if MATERIALX_VERSION_INDEX > MATERIALX_GENERATE_INDEX(1, 38, 8)
+    REQUIRE(true);
+#else
+    REQUIRE(false);
+#endif
 }
 
 TEST_CASE("String utilities", "[coreutil]")

